controllers/mattermost/mattermost/controller.go

```diff
@@ -144,10 +144,18 @@ func (r *MattermostReconciler) Reconcile(ctx context.Context, request ctrl.Reque
 		return reconcile.Result{}, err
 	}
 
-	err = r.checkMattermost(mattermost, dbConfig, fileStoreConfig, &status, reqLogger)
-	if err != nil {
-		r.updateStatusReconcilingAndLogError(mattermost, status, reqLogger, err)
-		return reconcile.Result{}, err
+	recStatus := reconcileStatus{
+		Status: false,
+		Error:  nil,
+	}
```

Comment:
Let's not include an error in the status, it is clearer if functions will return `(reconcileStatus, error)`.

Also, we can be a bit more specific here, perhaps instead of `Status` field, let's call it something like `ResourcesReady` or `Requeue` depending on what value we want to use as a default.

We can also remove this initialization and initialize a variable directly from the function.